compiler/rustc_lint/src/precedence.rs

```diff
@@ -0,0 +1,132 @@
+use rustc_ast::token::LitKind;
+use rustc_ast::{BinOpKind, Expr, ExprKind, MethodCall, UnOp};
+use rustc_span::source_map::Spanned;
+
+use crate::lints::{PrecedenceDiag, PrecedenceUnarySuggestion, PrecedenceUnwarySuggestion};
+use crate::{EarlyContext, EarlyLintPass, LintContext};
+
+// List of functions `f(x)` where `f(-x)=-f(x)` so the
+// precedence doens't matter.
+const ALLOWED_ODD_FUNCTIONS: [&str; 14] = [
```

Comment:
Personally I'd prefer it to warn consistently, even if the semantics of the function mean it doesn't matter. (a) that seems more consistent and less surprising, and (b) I am writing tests which ensure that `sin` behaves correctly, and I care about which sign the argument actually has.

Reply:
Not warn on odd method calls was asked in https://github.com/rust-lang/rust-clippy/issues/4892, except for the obvious "the precedence doesn't matter" with them, no other reasons were given, but the consistency issue seems higher to me.

Anyway I agree, this behavior is unexpected and weird, I've removed this logic, we can always re-add it later if people really want it.